src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -143,7 +143,7 @@ private unsafe int GetHashCodeOfStringCore(ReadOnlySpan<char> source, CompareOpt
 
                 byte[]? borrowedArr = null;
                 Span<byte> span = sortKeyLength <= 512 ?
-                    stackalloc byte[512] :
+                    stackalloc byte[sortKeyLength] :
```

Comment:
Why did you change this?

Reply:
To allocate exactly as much memory on the stack as required, not more.

Is there a reason I should not do that?

Reply:
@adamsitnik Allocating variable sized buffers via stackalloc tends to produce bigger and slower code. (It is the case with `ClearInitLocals` rewriter that we use for CoreCLR and framework - it is the other way around without the rewriter.)

Also, the variable sized stackalloc buffers have a stackoverflow / denial of service potential - they require more scrutiny for codereviews.

Reply:
@jkotas big thanks for the full explanation, I have reverted the change.